jdaviz/configs/default/plugins/plot_options/tests/test_plot_options.py

```diff
@@ -172,3 +176,57 @@ def test_user_api(cubeviz_helper, spectrum1d_cube):
     # toggle contour (which has a spinner implemented)
     po.contour_visible = True
     assert po._obj.contour_spinner is False
+
+
+@pytest.mark.filterwarnings('ignore')
+def test_stretch_spline(imviz_helper):
+    image_1 = NDData(make_4gaussians_image(), unit=u.nJy)
+    # Load the test data into imviz
+    imviz_helper.load_data(image_1)
+    po = imviz_helper.plugins['Plot Options']
+
+    # Configure initial stretch options and select "Spline" function
+    with po.as_active():
+        po.stretch_vmin.value = 10
+        po.stretch_vmax.value = 100
+        po.stretch_curve_visible = True
+        po.stretch_function = "Spline"
+
+    # Retrieve knots data from the generated histogram
+    scatter_obj = po._obj.stretch_histogram.marks["stretch_knots: NDData[DATA] (DATA)"]
+    knots_x = scatter_obj.x
+    knots_y = scatter_obj.y
+
+    # Expected knots based on initial stretch settings
+    expected_x = np.array([10., 19., 28., 73., 100.])
+    expected_y = np.array([0., 0.05, 0.3, 0.9, 1.])
```

Comment:
Not necessary to cast as `np.array`.
```suggestion
    expected_x = [10., 19., 28., 73., 100.]
    expected_y = [0., 0.05, 0.3, 0.9, 1.]
```